packages/hooks/src/useP2PAdvertiserPaymentMethods.ts

```diff
@@ -0,0 +1,77 @@
+import { useCallback, useMemo } from 'react';
+import { useFetch, useInvalidateQuery, useRequest } from '@deriv/api';
+import { useStore } from '@deriv/stores';
+
+const type_to_icon_mapper = {
+    bank: 'IcCashierBankTransfer',
+    other: 'IcCashierOther',
+    ewallet: 'IcCashierEwallet',
+};
+
+/** A custom hook to fetch, create, update, and delete p2p advertiser payment methods */
+const useP2PAdvertiserPaymentMethods = (handleMutationError?: (error?: unknown) => void) => {
+    const invalidate = useInvalidateQuery();
+    const { client } = useStore();
+    const { is_authorize } = client;
+
+    /** Make a request to p2p_advertiser_payment_methods from API and onSuccess it will invalidate the cached data  */
+    const { mutate } = useRequest('p2p_advertiser_payment_methods', {
+        onSuccess: () => {
+            invalidate('p2p_advertiser_payment_methods');
+        },
+        onError: error => {
+            if (error && handleMutationError) handleMutationError(error);
+        },
```

Comment:
```suggestion
```
@farhan-nurzi-deriv We don't need this, Just returning the stuff from `useRequest` should be enough to let the component handle these cases.

Reply:
The problem was useRequest didnt catch the error. I have tried to access it from component and it returns `null`. But, i will try to check it again 😄

Reply:
@farhan-nurzi-deriv `error` is `undefined` by default, You need to check it in a `useEffect`.

I'm free we can check it together if needed 🙇🏻

Reply:
Thanks @farzin-deriv , maybe i will continue on it later or tomorrow. I will comeback to you if i still havent figure it out 😄